### PR TITLE
[docs] fix findFork comment

### PR DESCRIPTION
**Pull request by r8921039**
The return value of findFork is an ancestor of the specified block only when specified block is _not_ on the active chain. When it is on the active chain, the return value is the specified block itself, not an ancestor of it.

### CODE REVIEW

**Comment by gmaxwell** [comment]
> Return height of the highest block on the chain that is or has an ancestor

"is or has an ancestor" doesn't make much sense to me.  The function is returning the highest common point, if the block is on the chain, that's obviously itself. The comment in chain.h is correct and I think not confusing: Find the last common block between this chain and a block index entry.

**Comment by r8921039** [comment]
Similarly, the following comment is true only if the locator is _not_ created by a block on the active chain. 

        //! Return height of the latest block common to locator and chain, which
        //! is guaranteed to be an ancestor of the block used to create the
        //! locator.
        virtual Optional<int> findLocatorFork(const CBlockLocator& locator) = 0;

**Comment by ryanofsky** [comment]
@r8921039, I wrote these comments and I can see how they are confusing. I wrote "ancestor" here when I really meant "ancestor-or-self" because it seemed less awkward (similar to how people write ⊂ instead of ⊆ in math for convenience, even though it can be less clear if you don't know the context).

I think "highest block on the chain that is or has an ancestor of the specified block" is hard to parse, though. Maybe it would be better to change these comments to:

- findFork: Return height of the specified block if it is on the chain, otherwise return the height of the highest block on chain that's an ancestor of the specified block, or nullopt if there is no common ancestor.
- findLocatorFork: Return height of the highest block on chain in common with the locator (which will either be the original the block used to create the locator, or one of its ancestors.)

**Comment by r8921039** [comment]
@ryanofsky Thank you very much for your response and suggestions. The PR is updated accordingly.

**Review by ryanofsky — APPROVED** [review]
utACK b837d66f6eba294ea4fd996bbdedc53ba9eab97b

**Comment by MarcoFalke** [comment]
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

**Comment by r8921039** [comment]
@MarcoFalke Yes sir, done as instructed.

**Comment by promag** [comment]
utACK c968780, however comment could be shorter.

**Review by ryanofsky — APPROVED** [review]
utACK c968780785c18bd6d0a8659c9e251ccf8fdc91dc. Only change since last review is squash

**Comment by r8921039** [comment]
Would someone be willing to give a bit more love to this PR... Thanks.

**Comment by ryanofsky** [comment]
This could just be merged I think. Documentation update with no change in meaning.

**Comment by r8921039** [comment]
@MarcoFalke Do we need more utACK to proceed? If you think so, I will try asking on IRC. Thanks.

**Comment by r8921039** [comment]
Thanks a lot guys! :)
